card_lib/cards.py

```diff
@@ -34,17 +36,50 @@ def __str__(self):
         return self.symbol
 
 
+ALL_SUITS = (Suit.HEART, Suit.DIAMOND, Suit.CLUB, Suit.SPADE)
+ALL_SUIT_SYMBOLS = [s.symbol for s in ALL_SUITS]
+
+
+def _symbol_to_suit(symbol):
+    """
+    Given a card symbol, find the matching card suit.
+    """
+    for one_suit in ALL_SUITS:
+        if one_suit.symbol == symbol:
+            return one_suit
+    raise UnknownSuitError('No suit found for symbol "{}"'.format(symbol))
+
+
 class Card(object):
 
     def __init__(self, pip, suit):
         self.pip = pip
-        self.suit = suit
+        if suit in ALL_SUIT_SYMBOLS:
+            self.suit = _symbol_to_suit(suit)
+        else:
+            self.suit = suit
+
+    def to_json(self):
+        """
+        Return a string containing a JSON representation of a card.
+        """
+        return '{{"suit":"{}", "pip":"{}"}}'.format(self.suit, self.pip)
+
+    @classmethod
+    def from_json(cls, json_str):
+        """
+        Loads a card from a valid JSON representation.
+        """
+        card = json.loads(json_str)
+        return cls(card['pip'], _symbol_to_suit(card['suit']))
```

Comment:
I see what this is doing, but am curious- Is this functionally equivalent to:
```
return Card(card['pip'], _symbol_to_suit(card['suit']))
```
? I understand that classmethods have access to class state as opposed to instance state, but in this case, you are not touching class state. Why would one choose ```cls``` over the constrcutor? Is it just shorthand?

Reply:
I've always avoided including the class name in its methods when at all possible, except `super()` calls, particularly when the `cls` is available - so that if the class is renamed or code is refactored, the class name is in as few places as possible.